src/mlpack/methods/logistic_regression/logistic_regression_function_impl.hpp

```diff
@@ -125,6 +125,41 @@ double LogisticRegressionFunction<MatType>::Evaluate(
     return -log(1.0 - sigmoid) + regularization;
 }
 
+/**
+ * Evaluate the logistic regression objective function given the estimated
+ * parameters for a given batch from a given point.
+ */
+template<typename MatType>
+double LogisticRegressionFunction<MatType>::Evaluate(
+                  const arma::mat& parameters,
+                  const size_t begin,
+                  const size_t batchSize) const
+{
+  // Calculating the regularization term.
+  const double regularization = 0.5 * lambda *
+      arma::dot(parameters.col(0).subvec(begin, batchSize - 1),
+              parameters.col(0).subvec(begin, batchSize - 1));
+
+  // Calculating the hypothesis that has to be passed to the sigmoid function.
+  const arma::vec exponents = parameters(0, 0) + predictors.t() *
+      parameters.col(0).subvec(begin, batchSize - 1);
+  // Calculating the sigmoid function values.
+  const arma::vec sigmoid = 1.0 / (1.0 + arma::exp(-exponents));
+
+  // Iterating for the given batch size from a given point
+  double result = 0.0;
+  for (size_t i = begin; i < batchSize; ++i)
```

Comment:
maybe I am wrong but shouldn't it be, i < begin + batchSize? Also should we do bound checks? maybe begin + batchSize is out of total examples for training.

Reply:
Oh yes, that's a mistake :sweat_smile: . It is supposed to be `i < begin + batchSize`. I'll fix it.